.github/workflows/main.yml

```diff
@@ -50,9 +50,15 @@ jobs:
 
   software-tests:
     name: Software Tests
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
```

Comment:
consider updating all the runners to `24.04` and switching line 17 to `22.04` (alternate build check)